rustls/src/error.rs

```diff
@@ -544,43 +546,59 @@ impl From<rand::GetRandomFailed> for Error {
     }
 }
 
-/// Any other error that cannot be expressed by a more specific [`Error`] variant.
-///
-/// For example, an `OtherError` could be produced by a custom crypto provider
-/// exposing a provider specific error.
-///
-/// Enums holding this type will never compare equal to each other.
-#[derive(Debug, Clone)]
-pub struct OtherError(pub Arc<dyn StdError + Send + Sync>);
+mod other_error {
+    #[cfg(feature = "std")]
+    use alloc::sync::Arc;
+    #[cfg(feature = "std")]
+    use core::fmt;
+    #[cfg(feature = "std")]
+    use std::error::Error as StdError;
+
+    use super::Error;
 
-impl PartialEq<Self> for OtherError {
-    fn eq(&self, _other: &Self) -> bool {
-        false
+    /// Any other error that cannot be expressed by a more specific [`Error`] variant.
+    ///
+    /// For example, an `OtherError` could be produced by a custom crypto provider
+    /// exposing a provider specific error.
+    ///
+    /// Enums holding this type will never compare equal to each other.
+    #[derive(Debug, Clone)]
+    pub struct OtherError(#[cfg(feature = "std")] pub Arc<dyn StdError + Send + Sync>);
+
+    impl PartialEq<Self> for OtherError {
+        fn eq(&self, _other: &Self) -> bool {
+            false
+        }
     }
-}
 
-impl From<OtherError> for Error {
-    fn from(value: OtherError) -> Self {
-        Self::Other(value)
+    impl From<OtherError> for Error {
+        fn from(value: OtherError) -> Self {
+            Self::Other(value)
+        }
     }
-}
 
-impl fmt::Display for OtherError {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.0)
+    #[cfg(feature = "std")]
+    impl fmt::Display for OtherError {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            write!(f, "{}", self.0)
```

Comment:
```suggestion
    impl fmt::Display for OtherError {
        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
            #[cfg(feature = "std")]
            write!(f, "{}", self.0)
            #[cfg(not(feature = "std"))]
            f.write_str("no further information available")
```

I think that should address the remaining gating in `impl fmt::Display for Error` above.